### PR TITLE
Fixes "Sparql\Client uses invalid parameters for Zend\Http\Client->setHeaders"

### LINKED ISSUE #305: ZF 2 HTTP Client not working
I needed zend http to do basic http authentication which the default http adapter doesn not seem to support.

According to this commit https://github.com/njh/easyrdf/commit/2de0339b3d1e5a2a557a51d8dc4220f935792661
It should work with zend http 2 but it calls the setHeaders method with invalid parameters.

```
Message: Invalid parameter headers passed
Stack trace:

#0 /var/www/openskos/vendor/easyrdf/easyrdf/lib/Sparql/Client.php(350): Zend\Http\Client->setHeaders('Accept', 'application/spa...')
```
The method setHeaders only accepts an array or Headers object.

### PR DESCRIPTION
**Pull request by k00ni**
It fixes the problem by using a proxy function, which uses a different function call (send or request) based on the client class.

Fixes #305

Ref:
* https://github.com/sweetyrdf/easyrdf/pull/10
* https://github.com/sweetyrdf/easyrdf/issues/9

### CODE REVIEW

File: test/EasyRdf/Sparql/ClientTest.php [hunk 1]
```diff
@@ -462,6 +462,82 @@ public function testAcceptHeaders()
         $this->assertContains('text/turtle', $types);
     }
 
+    /**
+     * Test for issue https://github.com/sweetyrdf/easyrdf/issues/9
```
Comment (njh) [inline test/EasyRdf/Sparql/ClientTest.php:466]: Please can you update this to:
https://github.com/easyrdf/easyrdf/issues/305

File: test/EasyRdf/Sparql/ClientTest.php [hunk 2]
```diff
@@ -462,6 +462,82 @@ public function testAcceptHeaders()
...
+
+        Http::setDefaultHttpClient($easyRdfHttpClient);
+
+        $this->sparql = new Client('http://localhost:8080/sparql');
+
+        $this->sparql->query('test');
+    }
+
+    /**
+     * Test for issue https://github.com/sweetyrdf/easyrdf/issues/9
```
Comment (njh) [inline test/EasyRdf/Sparql/ClientTest.php:502]: Please can you update this to:
https://github.com/easyrdf/easyrdf/issues/305

File: test/EasyRdf/Sparql/ClientTest.php [hunk 3]
```diff
@@ -462,6 +462,82 @@ public function testAcceptHeaders()
...
+        // client
+        $easyRdfHttpClient = $this->getMockBuilder(\Zend\Http\Client::class)->disableOriginalConstructor()->getMock();
+        $easyRdfHttpClient->method('send')->willReturn($response);
+        $easyRdfHttpClient
+            ->expects($this->exactly(1))
+            ->method('setHeaders')
+            /*
+             * we only check if an array with 'Accept' as key was given. ignore the value,
+             * because it may vary between PHP versions.
+             * FYI: https://github.com/sweetyrdf/easyrdf/pull/10/commits/aab3af54aa0064203f61f0046f0e16cafaeacdc7
```
Comment (njh) [inline test/EasyRdf/Sparql/ClientTest.php:526]: Please can you update this reference?

File: test/EasyRdf/Sparql/ClientTest.php (hunk 1 above)
Comment (k00ni) [reply test/EasyRdf/Sparql/ClientTest.php]: Done.

File: test/EasyRdf/Sparql/ClientTest.php (hunk 2 above)
Comment (k00ni) [reply test/EasyRdf/Sparql/ClientTest.php]: Done.

File: test/EasyRdf/Sparql/ClientTest.php (hunk 3 above)
Comment (k00ni) [reply test/EasyRdf/Sparql/ClientTest.php]: I removed it, because there is no further information in the commits related to this PR.

**Comment by k00ni** [comment]
Files adapted. I also adapted the name of the tests (`9` => `305`)

**Comment by k00ni** [comment]
Ping :)

**Comment by njh** [comment]
Note that Scrutinizer has reported:

>4 issues (4 major) were introduced and 5 issues were fixed.

**Review by njh — APPROVED** [review]

**Comment by k00ni** [comment]
Scrutinizer reports that these issues were introduced by you (@njh) and @indeyets :)

Ref: https://scrutinizer-ci.com/g/easyrdf/easyrdf/inspections/25126edc-e928-4ee4-bdce-68f8ec6af93f/issues/files/lib/Sparql/Client.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

But regardless, we should keep an eye on them.

**Comment by njh** [comment]
Hah, oops!

**Pull request merged**